src/feed.js

```diff
@@ -66,10 +109,85 @@ export default class WebsiteFeed {
         logger.info(`done pulling items from ${this.feedURL}`);
     }
 
+    // Useful for debugging.
+    async printItems() {
+        logger.info(`printing items from ${this.feedURL}`);
+
+        const parser = new Parser();
+        const feed = await parser.parseURL(this.feedURL);
+
+        feed.items.reverse();
+
+        feed.items.forEach((item) => {
+            console.log("\n");
+            console.log(JSON.stringify(item, null, 2));
+        });
+
+        logger.info(`done printing items from ${this.feedURL}`);
+    }
+
     async getItems() {
         if (this.items.length === 0) {
             await this.pullItems();
         }
         return this.items;
     }
 }
+
+export class VideoFeed extends WebsiteFeed {
+    constructor(feedURL, limit = 20) {
+        super(feedURL, limit);
+    }
+
+    async pullItems() {
+        logger.info(`pulling items from ${this.feedURL}`);
+
+        const parser = new Parser({
+            customFields: {
+                item: [["media:group", "media:group"]],
+            },
+        });
+        const feed = await parser.parseURL(this.feedURL);
+
+        // Make sure the latest post is first.
+        feed.items.reverse();
+
+        if (feed.items.length > this.limit) {
+            logger.debug(`using for-loop population for ${this.feedURL}`);
+
+            let newList = [];
+            for (let i = 0; i < this.limit; i++) {
+                newList.push(transformVideoItem(feed.items[i]));
+            }
+            this.items = [...newList];
+        } else {
+            logger.debug(`using map-spread population for ${this.feedURL}`);
+            this.items = [
+                ...feed.items.map((item) => transformVideoItem(item)),
+            ];
+        }
```

Comment:
I wonder if this could be simplified a little bit by using a slice. It would seem to me that you could do something like

``` javascript
this.items = [
    ...feed.items.slice(0, this.limit).map((item) => transformVideoItem(item)),
];
```

I'm probably missing something here, but it seems like it should work.

Reply:
That might work. I remember having issues with this section of the code for some reason and finally just giving up once I got it working with this solution.